### ISSUE TITLE
bug: Research Data Archive links are broken by new naming convention

### ISSUE THREAD

**Issue by kdraeder** [NCAR/DART#755]
### Describe the bug
Here's the issue I opened in the RDA Support site:

> I tried to look at a dataset I referenced in a paper (https://rda.ucar.edu/ds277.7/), but that leads “Page not found”.
> 
> Searching in the RDA Search tool for “ds277.7” yields no links.
> 
> When I browsed for the data set I got one which was close (d277008), but not the actual set (See attached.  I don’t know why the one I wanted was excluded).  Seeing the unfamiliar name format made me suspect that names had been changed.
> 
> It occurred to me to look in the News, and I see there that the RDA has a new naming convention “d277007”.  It seems that there should be a forwarding link from the old name to the new name, or at least a message to users that they need to translate the name.  If something like this is not implemented, then all the existing references and links to the RDA datasets will be broken, unless each user debugs the problem.  At least some of them won’t bother.

### Which model(s) are you working with?
No models; documentation

### Version of DART      
Which version of DART are you using? 
All, up through the latest release (2024-10-15).

### Have you modified the DART code?   
No  
The solution should be implemented in the RDA; links from old web pages to the new ones.
Or at least a warning that user's need to translate old names to a new format.

**Comment by kdraeder** [comment]
The RDA Support team answered:
 

> Douglas Schuster
> 15/Oct/24 1:55 PM
> 
> Hi Kevin Raeder, Thanks for the heads up on this issues and as we discussed during the fire alarm, I'll bring this up with our team and see what we can do are there are many publications that likely point to legacy RDA and even DSS links.  Also, please use the DOI link when you reference datasets in future publications as a best practice since these will always be set to resolve to the correct landing page associated with a referenced dataset. For example, In this case that link is: https://doi.org/10.5065/EM0T-1D34.  -Doug

I'll look through the DART web pages for 'RDA' and try to replace them with DOIs.
I don't know if I'll be able to efficiently find RDA in the publications, and if I do, how to fix them.

**Comment by kdraeder** [comment]
Bob Dattore commented:

> Hi Doug, Kevin,
> 
> We have had redirects from old ID to new ID in place and working since the changeover. The reason Kevin's link didn't work is because it is missing the "/datasets" part of the URL before the dataset ID, and it wouldn't have worked under the old ID system either. The correct URLs for that dataset are:
> 
> https://rda.ucar.edu/datasets/ds277.7
> and
> https://rda.ucar.edu/datasets/d277007

Hi Doug and Bob,

thanks for taking this on so quickly!  And for the DOI link.  I'll try to find ways to attach it and other DOIs to my publications.  And we can use them in our online docs too.

I checked the Wayback Machine at web.archive.org and see that the address for ds277.7 that I used in my paper was 'crawled' 18 times between Dec. 2019 and Dec. 2022, which brackets the period during which I wrote the paper.  I can't say whether the address I used was a link to a datasets/ds277.7, but it's highly unlikely that I just made up the address, so I must have copied it from somewhere I believed to be authoritative.  The 'datasets' address was crawled 73 times between 2013 and 2024, so it seems to be the more robust address.  Does it make sense to make forwarding links from the old format addresses that don't have 'datasets' in them?  For example; https://rda.ucar.edu/ds277.7 -> https://rda.ucar.edu/datasets/d277007

The situation is the same for my most important dataset; d345000.

Kevin

**Comment by kdraeder** [comment]
The RDA folks marked this as resolved and closed it.  
Here's my last entry in the issue.

---------------------------
We can be done with this issue, but I'd like there to be a summary of the status for future reference.

Addresses of the form https://rda.ucar.edu/ds277.7 worked in the past (~ 2019-2022). It no longer works, and won't be linked to the current address, so people will need to search for the dataset in the RDA site. Entering the old dataset name in the RDA search does find the dataset with the new name format, so that's a reasonable path for them to navigate.

A more common form was https://rda.ucar.edu**/datasets**/ds277.7. That form is linked to the current form (.../d277007), so that transition is seamless.

**Comment by hkershaw-brown** [comment]
> I'll look through the DART web pages for 'RDA' and try to replace them with DOIs.

Hi Kevin, did you find any RDA links in https://github.com/NCAR/DART.git  (docs.dart.ucar.edu)
or the DART website (dart.ucar.edu)

**Comment by kdraeder** [comment]
No, I forgot that I volunteered to do that, and didn't review the issue before I closed it.
Thanks for reminding me.

**Comment by hkershaw-brown** [comment]
fixed in #790
